### PR TITLE
add baseUrl option for GitHub to support enterprise installations.

### PR DESCRIPTION
**Pull request by jasonbellamy**

### CODE REVIEW

**Comment by plouc** [comment]
@jasonbellamy, thanks for spotting this out!
I think the config.api.github.baseUrl could be moved to the `buildApiRequest` function and the default value handled directly in the client, as it will require less boilerplate configuration.

```javascript
var apiBaseUrl = config.api.github.baseUrl || 'https://api.github.com';

function buildApiRequest(path) {
    var req = request.get(apiBaseUrl + path);
    …
````

**Comment by jasonbellamy** [comment]
Just let me know if you'd like to take care of updating the Wiki with the new `baseUrl` property or if you want me to take care of that.

File: src/ext/github/client.js [hunk 1]
```diff
@@ -5,8 +5,9 @@ require('superagent-bluebird-promise');
 
 var config = require('./../../../config');
 
-function buildApiRequest(url) {
-    var req = request.get(url);
+function buildApiRequest(path) {
+    var url = config.api.github.baseUrl || 'https://api.github.com';
```
Comment (plouc) [inline src/ext/github/client.js]: @jasonbellamy sorry for my code snippet proposal, I think the `url` var should stay outside the `buildApiRequest` function

**Comment by plouc** [comment]
For the Wiki update, if you want to do it, you're welcome!

File: src/ext/github/client.js (hunk 1 above)
Comment (jasonbellamy) [inline src/ext/github/client.js]: Even though it's not used or needed outside of that function?

**Comment by plouc** [comment]
@jasonbellamy, you're right, merged

**Comment by jasonbellamy** [comment]
Glad I could help! :+1:

**Pull request merged**